game_patch/multi/server.cpp

```diff
@@ -657,6 +665,26 @@ CodeInjection multi_limbo_init_injection{
     },
 };
 
+CallHook<void(const char* filename)> multi_change_level_call_hook{
+    0x0046E89C,
+    [](const char* filename){
+        if (rf::is_dedicated_server && g_additional_server_config.randomize_rotation) {
+            std::srand(static_cast<unsigned int>(std::time(nullptr)));
+            // select a random level index from the rotation, excluding the currently loaded level
+            int rand_level_index = rf::netgame.current_level_index;
```

Comment:
this logic (selection of random map with avoiding of repetition) is implemented in two places (for command/vote and here). This is a bad practice. You should make a function and reuse in both places.